allennlp/modules/seq2vec_encoders/cnn_encoder.py

```diff
@@ -0,0 +1,119 @@
+        This specifies both the number of convolutional layers we will create and their sizes.  The
+        default of (2, 3, 4, 5) will have four convolutional layers, corresponding to encoding
+        ngrams of size 2 to 5 with some number of filters.
+    conv_layer_activation: str, optional (default='relu')
+    """
+    def __init__(self,
+                 embedding_dim: int,
+                 num_filters: int,
+                 ngram_filter_sizes: Tuple[int, ...] = (2, 3, 4, 5),  # pylint: disable=bad-whitespace
+                 output_dim: Optional[int] = None) -> None:
+        super(CnnEncoder, self).__init__()
+        self._embedding_dim = embedding_dim
+        self._num_filters = num_filters
+        self._ngram_filter_sizes = ngram_filter_sizes
+        self._output_dim = output_dim
+
+        self._convolution_layers = [Conv1d(in_channels=self._embedding_dim,
+                                           out_channels=self._num_filters,
+                                           kernel_size=ngram_size)
+                                    for ngram_size in self._ngram_filter_sizes]
+        for i, conv_layer in enumerate(self._convolution_layers):
+            self.add_module('conv_layer_%d' % i, conv_layer)
+
+        maxpool_output_dim = self._num_filters * len(self._ngram_filter_sizes)
+        if self._output_dim:
+            self.projection_layer = Linear(maxpool_output_dim, self._output_dim)
+        else:
+            self.projection_layer = None
+            self._output_dim = maxpool_output_dim
+
+    @overrides
+    def get_output_dim(self) -> int:
+        return self._output_dim
+
+    def forward(self, tokens: torch.Tensor):  # pylint: disable=arguments-differ
+        # Our input is expected to have shape `(batch_size, num_tokens, embedding_dim)`.  The
+        # convolution layers expect input of shape `(batch_size, in_channels, sequence_length)`,
+        # where the conv layer `in_channels` is our `embedding_dim`.  We thus need to transpose the
+        # tensor first.
+        tokens = torch.transpose(tokens, 1, 2)
+
+        # Each convolution layer returns output of size `(batch_size, num_filters, pool_length)`,
+        # where `pool_length = num_tokens - ngram_size + 1`.  We then do an activation function
+        # (currently hard-coded to RELU), then do max pooling over each
+        # filter for the whole input sequence.  Because our max pooling is simple, we just use
+        # `torch.max`.  The resultant tensor of has shape `(batch_size, num_conv_layers *
+        # num_filters)`, which then gets projected using the projection layer, if requested.
+        # TODO(mattg): allow specifying the activation function to use here.
+        activation = torch.nn.functional.relu
+
+        filter_outputs = [torch.max(activation(convolution_layer(tokens), inplace=True),
+                                    dim=2)[0].squeeze(dim=2)  # not really sure why max isn't squeezing...
```

Comment:
The inplace argument is for the activation, not the max.  I'm not sure what you're suggesting would be more readable, as there are unmatched parentheses, and you're calling max on tokens, not the result of the activation function.